### DISCUSSION TITLE
Is there an ideal pi model to choose?

### DISCUSSION THREAD

**Discussion by RyanParsley** [cardonabits/haxo-hw#27] [General]
I see that the audio out allows for the support of a pi zero. Assuming you had access to any model, is zero the best pi for the job? Nice that it's smaller, but is it powerful enough? 

Also, there's like 3 pi zeros at the moment (zero, zero w and zero 2 w) are they all up to the task?

**Comment by jcard0na** [comment]
Hi Ryan,

Good questions all!

1. Which PI is best?  If you plan to do some development, (i.e. make code changes and recompile them natively), I'd go with a Raspberry Pi 3 or 4.  Compiling on the Pi Zero can take several minutes per build, which is only OK if you do that very infrequently or not at all.  If your goal is just to play, the Pi Zero works very well.  It takes close to a minute to boot, but after that, the playing performance is the same as the RPI3 or 4.

2. Of the 3 Pi Zeros, which one to choose?  That's easy: the cheapest you can get.  The Zero is the cheapest but they are impossible to find these days.  The next best would be the Pi Zero W, which you can find everywhere, even on [our crowdfunding page](https://www.crowdsupply.com/cardona-bits/haxophone).  As for the Zero 2, I have not tried it, but everything that I've read indicates that it should work as well or better than the old Zero.

Cheers,

**Reply by RyanParsley** [reply]
That works out good for me because I happen to have a couple cheap zeros around that I keep not wanting to use because of the lack of connectivity :D
